src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs

```diff
@@ -104,37 +104,33 @@ public IBindingProvider BindToGenericValueProvider<TAttribute>(Func<TAttribute,
         /// <typeparam name="TAttribute">Type of binding attribute on the user's parameter.</typeparam>
         /// <typeparam name="TMessage">'core type' for the IAsyncCollector.</typeparam>
         /// <param name="buildFromAttribute">Function to allocate the collector object given a resolved instance of the attribute.</param>
-        /// <param name="buildParameterDescriptor">An optional function to create a specific ParameterDescriptor object for the dashboard. If missing, a default ParameterDescriptor is created. </param>
-        /// <param name="postResolveHook">An advanced hook for translating the attribute. </param>
         /// <returns>A binding provider that applies these semantics.</returns>
         public IBindingProvider BindToAsyncCollector<TAttribute, TMessage>(
-            Func<TAttribute, IAsyncCollector<TMessage>> buildFromAttribute, 
-            Func<TAttribute, ParameterInfo, INameResolver, ParameterDescriptor> buildParameterDescriptor = null,
-            Func<TAttribute, ParameterInfo, INameResolver, Task<TAttribute>> postResolveHook = null)
+            Func<TAttribute, IAsyncCollector<TMessage>> buildFromAttribute)
```

Comment:
The problem is that when you add the rule, you can't refer to generic parameters (T), all types must be concrete. So we use a OpenType as a concrete placeholder for a generic parameter T. Ie, you'd add a rule as: 
    IConverter<CloudTable, IQueryable<OpenType>>.  
But your actual implementation uses a generic T instead of the concrete class OpenType.  Nobody uses OpenType in their implementations.